src/main/java/com/internhub/data/positions/scrapers/strategies/impl/PositionBFSMTStrategy.java

```diff
@@ -0,0 +1,84 @@
+package com.internhub.data.positions.scrapers.strategies.impl;
+
+import com.google.common.collect.Lists;
+import com.internhub.data.models.Company;
+import com.internhub.data.models.Position;
+import com.internhub.data.pages.Page;
+import com.internhub.data.positions.extractors.PositionExtractor;
+import org.openqa.selenium.WebDriver;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+/**
+ * Had to do something like this to escape the schedule executor service task
+ */
+class ScheduledDelayFinishedException extends RuntimeException {
+}
+
+public class PositionBFSMTStrategy extends PositionBFSStrategy {
+    private final ScheduledExecutorService scheduler;
+
+    public PositionBFSMTStrategy(WebDriver driver) {
+        super(driver);
+        scheduler = Executors.newScheduledThreadPool(1);
+    }
+
+    @Override
+    public List<Position> fetch(Company company, List<String> initialLinks) {
+        List<Position> results = Lists.newArrayList();
+        PriorityQueue<Candidate> candidates = new PriorityQueue<>(new CandidateComparator());
+        Set<String> visited = new HashSet<>(initialLinks);
+
+        final AtomicInteger totalLinks = new AtomicInteger(0);
+        candidates.addAll(initialLinks.stream()
+                .map((link) -> new Candidate(link, 1))
+                .collect(Collectors.toList()));
+
+        ScheduledFuture future = scheduler.scheduleWithFixedDelay(
+                () -> {
+                    if (!candidates.isEmpty() && totalLinks.get() < MAX_TOTAL_LINKS) {
+                        Candidate candidate = candidates.poll();
+                        logger.info(String.format(
+                                "[%d/%d] Visiting %s (depth = %d) ...",
+                                totalLinks.getAndIncrement(), MAX_TOTAL_LINKS, candidate.link, candidate.depth));
+
+                        processCandidate(company, candidate, candidates, visited, results);
+                    } else {
+                        throw new ScheduledDelayFinishedException();
```

Comment:
This is to break out of the scheduled task. Exceptions are slow so this isn't ideal.